webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -4466,6 +4468,7 @@ var CURRENT_GROUP_DATA={};
                 clearTimeout(MCK_TRIGGER_MSG_NOTIFICATION_PARAM);
                 MCK_MAINTAIN_ACTIVE_CONVERSATION_STATE && params.isGroup && params.tabId && KommunicateUtils.setItemToLocalStorage("mckActiveConversationInfo",{groupId:params.tabId, "appId":appOptions.appId, "userId":userId});
                 var currTabId = $mck_msg_inner.data('mck-id');
+                ACTIVE_TAB_ID = params.tabId || currTabId;
```

Comment:
@ab14bhardwaj any other way to figure out the active tab id? so that we don't have to maintain the state of it and hence avoid any possible errors.

Reply:
If not maintain then we have to get it from the UI from some other way. I guess maintaining the state would have been better rather than doing *$mck_msg_inner.data('mck-id');* in every function which we're doing currently. But no problem I will check for an alternate solution.

Reply:
problem with having one single truth at multiple places is that then both should be syncd, even if one of them is missed somewhere by someone then bugs will appear. One way to have it store on the variable as long with data is to have a single function which does both the things ie storing it to ACTIVE_TAB_ID and storing it in data, and then all the code should call that common function without the need to set the data('mck-id'). this way it will be ensured that both are in sync always.